flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorAlignmentTest.java

```diff
@@ -377,4 +424,9 @@ private void assertLatestWatermarkAlignmentEvent(int subtask, long expectedWater
         assertThat(events.get(events.size() - 1))
                 .isEqualTo(new WatermarkAlignmentEvent(expectedWatermark));
     }
+
+    private void assertHasNoMoreSplits(int subtask, boolean expected) {
```

Comment:
method name is `assertHasNoMoreSplits`, but it triggers `signalNoMoreSplits` as well.

I think `sourceCoordinator.getContext().signalNoMoreSplits(subtask);` should be moved out of this method.

Also, `assertHasNoMoreSplits` is only called once, and the method only has one line. I'm not sure whether we need extract it as a separate method. Caller can call `assertThat(sourceCoordinator.getContext().hasNoMoreSplits(subtask)).isEqualTo(expected);` directly.